webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ArchiveApi.scala

```diff
@@ -1054,6 +1055,50 @@ trait SaveArchiveService {
   def save(archive: PolicyArchive, mergePolicy: MergePolicy, actor: EventActor): IOResult[Unit]
 }
 
+final case class TechniqueArchiveMod(added: Chunk[ResourceFile], updated: Chunk[ResourceFile])
+
+object SaveArchiveServicebyRepo {
+
+  /*
+   * Create the resource diff between the archive technique and the corresponding path
+   * (assumed to be the technique base directory, ie ../category/1.0/techniqueId).
+   * If file does not exists, everything is marked as "new", including the base directory.
+   */
+  def buildDiff(t: TechniqueArchive, techniqueBaseDirectory: File): IOResult[Chunk[ResourceFile]] = {
+    for {
+      // path ot 't' are relative to technique. Files we need to add as new in the end
+      addedRef <- Ref.make(Chunk.fromIterable(t.files.map(_._1)))
+      diffRef  <- Ref.make(Chunk[ResourceFile]())
+      // get all file path (relative to technique dir)
+      existing <- IOResult.attempt {
+                    if (techniqueBaseDirectory.exists) {
+                      techniqueBaseDirectory
+                        .collectChildren(!_.isDirectory) // we don't have directories in the archive, only files
+                        .toList
+                        .map(_.pathAsString.replaceFirst("\\Q" + techniqueBaseDirectory.pathAsString + "/\\E", ""))
```

Comment:
we must quote the `replaceFirst` because path can contains `.` and `+` and other chars that would be interpreted as regex directives while we just want to cut the fisrt few path segments